transport/src/main/java/io/scalecube/transport/TransportEndpoint.java

```diff
@@ -118,43 +112,74 @@ public static InetSocketAddress parseSocketAddress(@CheckForNull String input) {
       throw new IllegalArgumentException();
     }
 
-    String host = Optional.fromNullable(matcher.group(1)).or(resolveLocalIpAddress());
-    if (isLocalhost(host)) {
-      host = resolveLocalIpAddress();
+    String host = parseHost(matcher.group(1));
+    int port = Integer.parseInt(matcher.group(2));
+    return InetSocketAddress.createUnresolved(host, port);
+  }
+
+  private static String parseHost(@CheckForNull String host) {
+    checkArgument(host != null);
+    checkArgument(!host.isEmpty());
+    return "localhost".equals(host) || "127.0.0.1".equals(host) ? getLocalIpAddress() : host;
+  }
+
+  /**
+   * Getting local IP4 address by the address of local host. <b>NOTE:</b> that returned IP4 address is expected to be a
+   * publicly visible IP4 address.
+   *
+   * @throws RuntimeException wrapped {@link UnknownHostException} in case when local host name couldn't be resolved
+   *         into an address; or when returned IP4 address belongs to loopback address family.
+   */
+  public static String getLocalIpAddress() {
```

Comment:
This method did not cover cases which were covered by IpAddressResolver (e.g. when you have both IPv6 and IPv4 interfaces on your machine). I propose to return back IpAddressResolver and use it.